File: spp/services/node/daemon_service.js
// daemon_service.js
console.log("Simulating heavy Node.js startup (takes 2 seconds)...");

let loaded = false;

// Simulate async heavy load
setTimeout(() => {
    loaded = true;
    console.log("Node.js heavy model loaded!");
}, 2000);

// We need a synchronous sleep just to simulate the module load blocking,
// or we just return an error if called before loaded.
// Since PolyglotBridge's ephemeral mode waits for the command to finish, 
// let's do a synchronous sleep so it's a true 1:1 comparison.
const start = Date.now();
while (Date.now() - start < 2000) {
    // Synchronous block to simulate heavy require() e.g. tfjs
}
console.log("Node.js module initialized.");

module.exports = {
    generate: function(prompt) {
        return `Node.js AI says: Hello! You asked: ${prompt}`;
    }
};

File: spp/services/perl/daemon_service.pl
package DaemonService;
use strict;
use warnings;
use Time::HiRes qw(sleep);

print "Simulating heavy Perl startup (takes 2 seconds)...\n";
sleep(2.0);
print "Perl module initialized.\n";

sub generate {
    my $prompt = shift;
    return "Perl AI says: Hello! You asked: $prompt";
}

1;
